cli/azd/internal/scaffold/bicep_env.go

```diff
@@ -0,0 +1,160 @@
+package scaffold
+
+import (
+	"fmt"
+	"strings"
+)
+
+func ToBicepEnv(env ResourceConnectionEnv) BicepEnv {
+	switch env.ResourceConnectionEnvType {
+	case ResourceConnectionEnvTypeServiceConnectorCreated:
+		return BicepEnv{
+			BicepEnvType: BicepEnvTypeServiceConnectorCreated,
+			Name:         env.Name,
+		}
+	case ResourceConnectionEnvTypePlainText:
+		return BicepEnv{
+			BicepEnvType: BicepEnvTypePlainText,
+			Name:         env.Name,
+			Value:        env.PlainTextValue,
+		}
+	case ResourceConnectionEnvTypeResourceSpecific:
+		value, ok := resourceSpecificBicepEnvValue[env.ResourceType][env.ResourceInfoType]
+		if !ok {
+			panic(unsupportedType(env))
+		}
+		if isSecret(env.ResourceInfoType) {
+			if isKeyVaultSecret(value) {
+				return BicepEnv{
+					BicepEnvType: BicepEnvTypeKeyVaultSecret,
+					Name:         env.Name,
+					SecretName:   secretName(env),
+					SecretValue:  unwrapKeyVaultSecretValue(value),
+				}
+			} else {
+				return BicepEnv{
+					BicepEnvType: BicepEnvTypeSecret,
+					Name:         env.Name,
+					SecretName:   secretName(env),
+					SecretValue:  value,
+				}
+			}
+		} else {
+			return BicepEnv{
+				BicepEnvType: BicepEnvTypePlainText,
+				Name:         env.Name,
+				Value:        value,
+			}
+		}
+	default:
+		panic(unsupportedType(env))
+	}
+}
+
+type BicepEnv struct {
+	BicepEnvType BicepEnvType
+	Name         string
+	Value        string
+	SecretName   string
+	SecretValue  string
+}
```

Comment:
This is environment variable used in bicep template file. It's OK that it's coupled with the template file.